src/components/Header.jsx

```diff
@@ -0,0 +1,129 @@
+import React, { useMemo } from "react";
```

Comment:
Overall proposal:

Each component has its own folder in `/src/components` named by component's name.
The file itself has index.jsx name

By that, you can create separate styles, constants, types files etc. for each component.
More cleaner way to manage huge amount of components.

<img width="158" alt="Screenshot 2024-12-10 at 19 20 17" src="https://github.com/user-attachments/assets/35cc4d44-b6b7-4d48-add9-6044df8125b0">